sql: nil pointer dereference on parsing a VIEW without PRIMARY KEY

## Problem

**Now, in current architecture**
```sql
VIEW v1 (
   fld1 int32
);
```
causes panic:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x1267c99]

goroutine 6 [running]:
testing.tRunner.func1.2({0x1452400, 0x1bf98d0})
	c:/Workspace/goinstall/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	c:/Workspace/goinstall/src/testing/testing.go:1548 +0x397
panic({0x1452400?, 0x1bf98d0?})
	c:/Workspace/goinstall/src/runtime/panic.go:914 +0x21f
github.com/voedger/voedger/pkg/parser.analyseView(0xc000351970, 0xc00009c330)
	c:/Workspace/voedger/pkg/parser/impl_analyse.go:163 +0x6d9
github.com/voedger/voedger/pkg/parser.analyse.func1({0x14eb720?, 0xc000351970?}, 0xc000351760?)
	c:/Workspace/voedger/pkg/parser/impl_analyse.go:51 +0xd4
github.com/voedger/voedger/pkg/parser.iterateContext.func1({0x14eb720, 0xc000351970})
	c:/Workspace/voedger/pkg/parser/utils.go:184 +0x36
github.com/voedger/voedger/pkg/parser.(*WorkspaceStmt).Iterate(0xc000351e40, 0xc0002aa4b0)
	c:/Workspace/voedger/pkg/parser/types.go:213 +0x5d
github.com/voedger/voedger/pkg/parser.iterateContext(0xc00009c330, 0x15c0d58)
	c:/Workspace/voedger/pkg/parser/utils.go:183 +0x76
github.com/voedger/voedger/pkg/parser.iterateContext.func1({0x14b0220, 0xc000351e40})
	c:/Workspace/voedger/pkg/parser/utils.go:192 +0xe5
github.com/voedger/voedger/pkg/parser.(*SchemaAST).Iterate(0xc00003f740, 0xc0002aa3d8)
	c:/Workspace/voedger/pkg/parser/types.go:82 +0x37
github.com/voedger/voedger/pkg/parser.iterateContext(0xc00009c180, 0x15c0d58)
	c:/Workspace/voedger/pkg/parser/utils.go:183 +0x76
github.com/voedger/voedger/pkg/parser.iteratePackage(...)
	c:/Workspace/voedger/pkg/parser/utils.go:170
github.com/voedger/voedger/pkg/parser.analyse(0xc0002d73a0, 0xc0003f14a0)
	c:/Workspace/voedger/pkg/parser/impl_analyse.go:36 +0x8e
github.com/voedger/voedger/pkg/parser.buildAppSchemaImpl({0xc0002d7380, 0x3, 0x15c04c0?})
	c:/Workspace/voedger/pkg/parser/impl.go:290 +0x218
github.com/voedger/voedger/pkg/parser.BuildAppSchema(...)
	c:/Workspace/voedger/pkg/parser/provide.go:41
github.com/voedger/voedger/pkg/vvm.buildSchemasASTs({0x167c088, 0xc000218bd0}, {0x1672ef8?, 0xc000047e00?})
...
```

- Problem (warm): I do not know what I did wrong

**Describe the solution you'd like**
a clear error message saying that I missed the primary key definition

--- a/pkg/parser/impl_analyse.go
+++ b/pkg/parser/impl_analyse.go
@@ -159,7 +159,9 @@ func analyseView(view *ViewStmt, c *iterateCtx) {
 	}
 	if view.pkRef == nil {
 		c.stmtErr(&view.Pos, ErrPrimaryKeyNotDefined)
+		return
 	}
+
 	for _, pkf := range view.pkRef.PartitionKeyFields {
 		index, ok := fields[string(pkf)]
 		if !ok {

--- a/pkg/parser/impl_test.go
+++ b/pkg/parser/impl_test.go
@@ -787,6 +787,17 @@ func Test_Views(t *testing.T) {
 		);
 	)
 	`, "file2.sql:4:4: reference to abstract table abc", "file2.sql:5:4: unexisting undefined")
+
+	f(`APPLICATION test(); WORKSPACE Workspace (
+		VIEW test(
+			fld1 int32
+		) AS RESULT OF Proj1;
+		EXTENSION ENGINE BUILTIN (
+			PROJECTOR Proj1 ON (Orders) INTENTS (View(test));
+			COMMAND Orders()
+		);
+	)
+	`, "file2.sql:2:3: primary key not defined")
 }
 
 func Test_Views2(t *testing.T) {

